discord/onboarding.py

```diff
@@ -0,0 +1,209 @@
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional, Set, List, Union
+
+from . import utils
+from .enums import OnboardingPromptType, try_enum
+from .utils import cached_slot_property
+
+__all__ = ('Onboarding', 'OnboardingPrompt', 'OnboardingPromptOption')
+
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .emoji import Emoji
+    from .guild import Guild
+    from .partial_emoji import PartialEmoji
+    from .role import Role
+    from .state import ConnectionState
+    from .threads import Thread
+    from .types.onboarding import (
+        Prompt as PromptPayload,
+        PromptOption as PromptOptionPayload,
+        Onboarding as OnboardingPayload,
+    )
+
+
+class OnboardingPromptOption:
+    """Represents an onboarding prompt option.
+
+    .. versionadded:: 2.2
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The ID of this prompt option.
+    guild: :class:`Guild`
+        The guild the onboarding prompt option is related to.
+    title: :class:`str`
+        The title of this prompt option.
+    description: Optional[:class:`str`]
+        The description of this prompt option.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`, :class:`str`]]
+        The emoji tied to this option. May be a custom emoji, or a unicode emoji.
+    channel_ids: Set[:class:`int`]
+        The IDs of the channels that will be made visible if this option is selected.
+    role_ids: Set[:class:`int`]
+        The IDs of the roles given to the user if this option is selected.
+    """
+
+    __slots__ = (
+        '_state',
+        '_cs_channels',
+        '_cs_roles',
+        'guild',
+        'id',
+        'title',
+        'description',
+        'emoji',
+        'channel_ids',
+        'role_ids',
+    )
+
+    def __init__(self, *, data: PromptOptionPayload, state: ConnectionState, guild: Guild) -> None:
+        self._state: ConnectionState = state
+        self.guild: Guild = guild
+        self.id: int = int(data['id'])
+        self.title: str = data['title']
+        self.description: Optional[str] = data['description']
+        self.emoji: Optional[Union[PartialEmoji, Emoji, str]] = self._state.get_reaction_emoji(data['emoji'])
+        self.channel_ids: Set[int] = {int(channel_id) for channel_id in data['channel_ids']}
+        self.role_ids: Set[int] = {int(role_id) for role_id in data['role_ids']}
+
+    def __repr__(self) -> str:
+        return f'<OnboardingPromptOption id={self.id} title={self.title}>'
+
+    @cached_slot_property('_cs_channels')
+    def channels(self) -> List[Union[GuildChannel, Thread]]:
+        """List[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The list of channels which will be set visible if this option is selected."""
```

Comment:
```suggestion
        """List[Union[:class:`abc.GuildChannel`, :class:`Thread`]]: The list of channels which will be made visible if this option is selected."""
```
For consistency with the other one.